_pytest/test_utf8_helpers.py

```diff
@@ -23,19 +23,23 @@ def test_decode_preserves_iterable_type():
     assert type(value_tuple) == type(decode_from_utf8(value_tuple))
 
 def test_decodes_utf8_string_to_unicode():
-    assert u'æøå' == decode_from_utf8(b'æøå')
+    assert u'æøå' == decode_from_utf8(b'æøå')  # noqa
```

Comment:
`b'æøå'` is a syntax error, so `# noqa` won't help as that is only a indication to linters. I'm not sure myself at the moment how to deal with this, will have to think about it.

Reply:
We are covered.  The try / except block keeps that line from ever executing on Python 3 and the noqa keeps the linters from complaining.

Reply:
What does [__six.b('æøå')__](http://six.readthedocs.io/#binary-and-text-data) do?

Reply:
Are you asking me? I've never used six :)

But it documents that it encodes in latin-1 on python 3, which we don't want to use. It made me realize that `b'...'` is equivalent to `'...'` in python 2 though, so we can just remove the b's to get rid of the syntax errors. However, that may not give the correct behavior in python 3, will have to check this.

Reply:
I guess these tests never made any sense in the first place. Go figure.

Something like `decode_from_utf8('æøå'.encode())` might make sense though. But remind me why we have a `decode_from_utf8()` in the first place.

Reply:
@auscompgeek: I do think the tests make sense, even though the `b` in front of the strings was redundant. But they just make sense in python 2, which they were written for, not in python 3 which handles unicode differently.

We have `decode_from_utf8` to recursively convert structures from bytestrings to unicode. All the strings we receive from weechat are bytestrings, but unicode strings are much more convenient and safer to work with. So we need to convert everything going from/to weechat.